### ISSUE TITLE
Periodically refresh the route with the latest traffic and ETAs

### ISSUE THREAD

**Issue by 1ec5** [mapbox/mapbox-navigation-ios#2284]
During turn-by-turn navigation, the SDK should periodically update the current route’s attributes and ETAs to reflect changing traffic conditions.

A specialized Directions API returns a sparse response that we’ll need to merge into the existing route and route progress models. This will require the ability to mutate the route model without recreating it from scratch. Currently most of the properties in Route, RouteLeg, etc. are constant. MapboxDirections.swift could define mutable subclasses, or define these types to structures that come with mutability built-in, or implement built-in methods for merging in updated data.

The route refreshing feature will complement two other forms of updates the SDK already performs: proactive rerouting periodically checks for a more optimal route than the current one, while ETA refreshing simply updates the ETA to reflect whether the user is keeping pace.

This feature has already been implemented on Android in mapbox/mapbox-navigation-android#1855.

/cc @mapbox/navigation-ios

**Comment by Udumft** [comment]
mapbox/mapbox-navigation-android#1855 seems to be outdated implementation. Rote Refreshing was updated on Android  in 1.0 in [Nav SDK #2570](https://github.com/mapbox/mapbox-navigation-android/pull/2570) to match [it's current API structure](https://github.com/mapbox/mapbox-java/tree/master/services-directions-refresh)

**Comment by 1ec5** [comment]
An implementation is moving in mapbox/mapbox-directions-swift#420 and #2366. Testing is currently blocked by #2574.
